### PR TITLE
feat: V2 batch update products in categories

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/types/src/workflow/product-category/index.ts [hunk 1]
```diff
@@ -14,3 +15,8 @@ export interface UpdateProductCategoryWorkflowInput {
   id: string
   data: UpdateProductCategoryDTO
 }
+
+export interface BatchUpdateProductsOnCategoryWorkflowInput
+  extends LinkMethodRequest {
+  id: string
+}
```
Comment (riqwan) [inline packages/types/src/workflow/product-category/index.ts]: There's also `LinkWorkflowInput`

```suggestion
export interface BatchUpdateProductsOnCategoryWorkflowInput
  extends LinkWorkflowInput {}
```

File: packages/types/src/workflow/product-category/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { LinkMethodRequest } from "../../common"
```
Comment (riqwan) [inline packages/types/src/workflow/product-category/index.ts]:
```suggestion
import { LinkWorkflowInput } from "../../common"
```

File: packages/core-flows/src/product/steps/batch-link-products-in-category.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      }
+    )
+
+    const productsWithUpdatedCategories = dbProducts.map((p) => {
+      if (toRemoveSet.has(p.id)) {
+        return {
+          id: p.id,
+          category_ids: (p.categories ?? [])
+            .filter((c) => c.id !== data.id)
+            .map((c) => c.id),
```
Comment (riqwan) [inline packages/core-flows/src/product/steps/batch-link-products-in-category.ts:36]: q: should we just pass categories and skip the song and dance between module and workflows?

```suggestion
          categories: (p.categories ?? [])
            .filter((c) => c.id !== data.id)
```

**Review by riqwan — APPROVED** [review]
lgtm!

File: packages/types/src/workflow/product-category/index.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/types/src/workflow/product-category/index.ts]: Nice, will use that

File: packages/core-flows/src/product/steps/batch-link-products-in-category.ts (hunk 3 above)
Comment (olivermrbl) [reply packages/core-flows/src/product/steps/batch-link-products-in-category.ts:36]: Could make sense. We can even go as far as typing the `categories` input as `categories: { id: string }[]`, and then we wouldn't have to normalize anything at either level.

Comment (riqwan) [reply packages/core-flows/src/product/steps/batch-link-products-in-category.ts:36]: I think this is how we do for some other modules as well

Comment (olivermrbl) [reply packages/core-flows/src/product/steps/batch-link-products-in-category.ts:36]: Keeping it as is for now and creating a ticket

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by sradevski — APPROVED** [review]

**Pull request merged**